docs/design_doc.md

```diff
@@ -0,0 +1,116 @@
+***Design Document***
```

Comment:
This design document is out of date. Think about including these comments directly into the code to explain each function. The design document should be reserved for high-level design discussion and proposed example code.

Reply:
Okay, will update this document.